File: sgi-webapp/src/app/core/models/csp/estado-solicitud.ts

import { marker } from '@biesbjerg/ngx-translate-extract-marker';
import { DateTime } from 'luxon';

export interface IEstadoSolicitud {
  /** Id */
  id: number;
  /** ID de la solicitud */
  solicitudId: number;
  /** Estado */
  estado: Estado;
  /** Fecha estado */
  fechaEstado: DateTime;
  /** Comentario */
  comentario: string;
}

export enum Estado {
  BORRADOR = 'BORRADOR',
  PRESENTADA = 'PRESENTADA',
  ADMITIDA_PROVISIONAL = 'ADMITIDA_PROVISIONAL',
  EXCLUIDA_PROVISIONAL = 'EXCLUIDA_PROVISIONAL',
  ALEGADA_ADMISION = 'ALEGADA_ADMISION',
  EXCLUIDA = 'EXCLUIDA',
  ADMITIDA_DEFINITIVA = 'ADMITIDA_DEFINITIVA',
  CONCECIDA_PROVISIONAL = 'CONCECIDA_PROVISIONAL',
  DENEGADA_PROVISIONAL = 'DENEGADA_PROVISIONAL',
  ALEGADA_CONCESION = 'ALEGADA_CONCESION',
  DESISTIDA = 'DESISTIDA',
  CONCECIDA = 'CONCECIDA',
  DENEGADA = 'DENEGADA',
}

export const ESTADO_MAP: Map<Estado, string> = new Map([
  [Estado.BORRADOR, marker(`csp.estado-solicitud.BORRADOR`)],
  [Estado.PRESENTADA, marker(`csp.estado-solicitud.PRESENTADA`)],
  [Estado.ADMITIDA_PROVISIONAL, marker(`csp.estado-solicitud.ADMITIDA_PROVISIONAL`)],
  [Estado.EXCLUIDA_PROVISIONAL, marker(`csp.estado-solicitud.EXCLUIDA_PROVISIONAL`)],
  [Estado.ALEGADA_ADMISION, marker(`csp.estado-solicitud.ALEGADA_ADMISION`)],
  [Estado.EXCLUIDA, marker(`csp.estado-solicitud.EXCLUIDA`)],
  [Estado.ADMITIDA_DEFINITIVA, marker(`csp.estado-solicitud.ADMITIDA_DEFINITIVA`)],
  [Estado.CONCECIDA_PROVISIONAL, marker(`csp.estado-solicitud.CONCECIDA_PROVISIONAL`)],
  [Estado.DENEGADA_PROVISIONAL, marker(`csp.estado-solicitud.DENEGADA_PROVISIONAL`)],
  [Estado.ALEGADA_CONCESION, marker(`csp.estado-solicitud.ALEGADA_CONCESION`)],
  [Estado.DESISTIDA, marker(`csp.estado-solicitud.DESISTIDA`)],
  [Estado.CONCECIDA, marker(`csp.estado-solicitud.CONCECIDA`)],
  [Estado.DENEGADA, marker(`csp.estado-solicitud.DENEGADA`)]
]);

File: sgi-csp-service/src/main/java/org/crue/hercules/sgi/csp/model/EstadoSolicitud.java
package org.crue.hercules.sgi.csp.model;

import java.time.LocalDateTime;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "estado_solicitud")
@Data
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class EstadoSolicitud extends BaseEntity {

  /**
   * Serial version
   */
  private static final long serialVersionUID = 1L;

  /**
   * Enumerado tipo estado de las solicitudes.
   *
   */
  public enum Estado {
    /** Borrador */
    BORRADOR,
    /** Presentada */
    PRESENTADA,
    /** Admitida provisional */
    ADMITIDA_PROVISIONAL,
    /** Excluida provisional */
    EXCLUIDA_PROVISIONAL,
    /** Alegada admisión */
    ALEGADA_ADMISION,
    /** Excluida */
    EXCLUIDA,
    /** Admitida definitiva */
    ADMITIDA_DEFINITIVA,
    /** Concedida provisional */
    CONCECIDA_PROVISIONAL,
    /** Denegada provisional */
    DENEGADA_PROVISIONAL,
    /** Alegada admisión */
    ALEGADA_CONCESION,
    /** Desistida */
    DESISTIDA,
    /** Concedida */
    CONCECIDA,
    /** Denegada */
    DENEGADA;
  }

  /** Id */
  @Id
  @Column(name = "id", nullable = false)
  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "estado_solicitud_seq")
  @SequenceGenerator(name = "estado_solicitud_seq", sequenceName = "estado_solicitud_seq", allocationSize = 1)
  private Long id;

  /** Solicitud */
  @Column(name = "id_solicitud", nullable = false)
  @NotNull
  private Long idSolicitud;

  /** Tipo estado solicitud */
  @Column(name = "estado", length = 50, nullable = false)
  @Enumerated(EnumType.STRING)
  @NotNull
  private Estado estado;

  /** Fecha. */
  @Column(name = "fecha_estado", nullable = false)
  @NotNull
  private LocalDateTime fechaEstado;

  /** Comentario */
  @Column(name = "comentario", length = 2000, nullable = true)
  @Size(max = 2000)
  private String comentario;

}